nzediegwu1/EventManager
# #156981666 Reimplement event/center posting to use client side image upload

#### What does this PR do?
Reimplement event/center posting to use client side image upload

#### Description of Task to be completed?
1. Modify addCenter component to implement clientside image upload before server API call
2. Modify addEvent component to implement clientside image upload before server API call
3. Implement functionality to hide sidebar on navigation click
4. Modify center controller to not use cloudinary file upload in center adding and update
5. Modify event controller to not use cloudinary file upload in center adding and update
6. Use query to pass and obtain publicId of event upon deletion
7. Modify error response in eventController, centerController, validator and authenticator to first delete uploaded image from cloudinary
8. Change public_id field in even and center models to publicId
9. Eliminate multer implementation for file-upload in event and center router
10. Change event and center data format to be sent from client from formData to json

#### How should this be manually tested?
on terminal, run command:
1. `npm run start`, then run:
2. `npm run client` (on another console)
3. login to the client app and add/edit event/center with image upload
4. check that transaction was successful without error

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories? (If applicable)
[#156981666](https://www.pivotaltracker.com/story/show/156981666)

#### Screenshots (If applicable)
None
